Source/WebSocket/hcwebsocket.cpp

```diff
@@ -101,25 +109,71 @@ try
         return E_INVALIDARG;
     }
 
-    auto httpSingleton = get_http_singleton(true);
-    if (nullptr == httpSingleton)
-        return E_HC_NOT_INITIALISED;
+    std::shared_ptr<websocket_connect_context> connectContext = std::make_shared<websocket_connect_context>();
+    connectContext->uri = uri;
+    connectContext->subProtocol = subProtocol;
+    connectContext->websocket = websocket;
+    connectContext->outerAsyncBlock = asyncBlock;
+    websocket_connect_context* rawConnectContext = static_cast<websocket_connect_context*>(shared_ptr_cache::store<websocket_connect_context>(connectContext));
 
-    auto connectFunc = httpSingleton->m_websocketConnectFunc;
-    if (connectFunc != nullptr)
+    HRESULT hr = XAsyncBegin(asyncBlock, rawConnectContext, reinterpret_cast<void*>(HCWebSocketConnectAsync), __FUNCTION__,
```

Comment:
why did you change this such that the connect is kicked off asynchronously?

Reply:
Ah, good catch.  This is wrong